ui/app/components/gas-customization/gas-price-chart/gas-price-chart.utils.js

```diff
@@ -0,0 +1,328 @@
+    setSelectedCircle({
+      chart,
+      newPrice,
+      closestLowerValueIndex,
+      closestLowerValue,
+      closestHigherValueIndex,
+      closestHigherValue,
+    })
+  } else {
+    hideDataUI(chart, cssId)
+  }
+}
+
+export function getAdjacentGasPrices ({ gasPrices, priceToPosition }) {
+  const closestLowerValueIndex = gasPrices.findIndex((e, i, a) => e <= priceToPosition && a[i + 1] >= priceToPosition)
+  const closestHigherValueIndex = gasPrices.findIndex((e, i, a) => e > priceToPosition)
+  return {
+    closestLowerValueIndex,
+    closestHigherValueIndex,
+    closestHigherValue: gasPrices[closestHigherValueIndex],
+    closestLowerValue: gasPrices[closestLowerValueIndex],
+  }
+}
+
+export function extrapolateY ({ higherY, lowerY, higherX, lowerX, xForExtrapolation }) {
+  const slope = (higherY - lowerY) / (higherX - lowerX)
+  const newTimeEstimate = -1 * (slope * (higherX - xForExtrapolation) - higherY)
+
+  return newTimeEstimate
+}
+
+
+export function getNewXandTimeEstimate ({ xMousePos, chartXStart, chartWidth, gasPrices, estimatedTimes }) {
+  const chartMouseXPos = xMousePos - chartXStart
+  const posPercentile = chartMouseXPos / chartWidth
+
+  const currentPosValue = (gasPrices[gasPrices.length - 1] - gasPrices[0]) * posPercentile + gasPrices[0]
+
+  const {
+    closestLowerValueIndex,
+    closestLowerValue,
+    closestHigherValueIndex,
+    closestHigherValue,
+  } = getAdjacentGasPrices({ gasPrices, priceToPosition: currentPosValue })
+
+  return !closestHigherValue || !closestLowerValue
+    ? {
+      currentPosValue: null,
+      newTimeEstimate: null,
+    }
+    : {
+      currentPosValue,
+      newTimeEstimate: extrapolateY({
+        higherY: estimatedTimes[closestHigherValueIndex],
+        lowerY: estimatedTimes[closestLowerValueIndex],
+        higherX: closestHigherValue,
+        lowerX: closestLowerValue,
+        xForExtrapolation: currentPosValue,
+      }),
+    }
+}
+
+export function hideDataUI (chart, dataNodeId) {
+  const overLayedCircle = d3.select(dataNodeId)
+  if (!overLayedCircle.empty()) {
+    overLayedCircle.remove()
+  }
+  d3.select('.c3-tooltip-container').style('display', 'none !important')
+  chart.internal.hideXGridFocus()
+}
+
+export function setTickPosition (axis, n, newPosition, secondNewPosition) {
+  const positionToShift = axis === 'y' ? 'x' : 'y'
+  const secondPositionToShift = axis === 'y' ? 'y' : 'x'
+  d3.select('#chart')
+    .select(`.c3-axis-${axis}`)
+    .selectAll('.tick')
+    .filter((d, i) => i === n)
+    .select('text')
+    .attr(positionToShift, 0)
+    .select('tspan')
+    .attr(positionToShift, newPosition)
+    .attr(secondPositionToShift, secondNewPosition || 0)
+    .style('visibility', 'visible')
+}
+
+export function appendOrUpdateCircle ({ data, itemIndex, cx, cy, cssId, appendOnly }) {
+  const circle = this.main
+    .select('.' + 'c3-selected-circles' + this.getTargetSelectorSuffix(data.id))
```

Comment:
Can this just be `'.c3-selected-circles' + this.getTargetSelectorSuffix(data.id)`?

Reply:
:+1:

Reply:
Done in 1f0cb6e